executor.py

```diff
@@ -63,18 +66,24 @@ def __init__(
         )
 
         self.logger = JinaLogger(self.metas.name)
+        if not self.tracer:
+            self.tracer = NoOpTracer()
 
     @requests(on='/index')
-    def index(self, docs: DocumentArray, **kwargs):
+    def index(self, docs: DocumentArray, tracing_context=None, **kwargs):
         """Index new documents
         :param docs: the Documents to index
         """
-        self._index.extend(docs)
+        with self.tracer.start_as_current_span(
+            'qdrant_index', context=tracing_context
+        ) as span:
+            self._index.extend(docs)
 
     @requests(on='/search')
     def search(
         self,
         docs: 'DocumentArray',
+        tracing_context=None,
```

Comment:
@samsja @JohannesMessner @alaeddine-13 For some reason I had to make the `tracing_context` as optional arugment otherwise the tests would fail. I'm not able to understand this because the [worker request handler](https://github.com/jina-ai/jina/blob/master/jina/serve/runtimes/worker/request_handling.py#L297) already has a default value of None.

The error is:
```
__________________________________________________ test_index ___________________________________________________

docs = <DocumentArray (length=6) at 140206597296848>, docker_compose = None

    def test_index(docs, docker_compose):
        qindex = QdrantIndexer(collection_name='test')
>       qindex.index(docs)

tests/unit/test_executor.py:58: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

executor_instance = <executor.QdrantIndexer object at 0x7f84646d5930>
args = (<DocumentArray (length=6) at 140206597296848>,), kwargs = {}

    @functools.wraps(fn)
    def arg_wrapper(
            executor_instance, *args, **kwargs
    ):  # we need to get the summary from the executor, so we need to access the self
>       return fn(executor_instance, *args, **kwargs)
E       TypeError: QdrantIndexer.index() missing 1 required positional argument: 'tracing_context'

venv/lib/python3.10/site-packages/jina/serve/executors/decorators.py:182: TypeError
============================================ short test summary info ============================================
FAILED tests/unit/test_executor.py::test_index - TypeError: QdrantIndexer.index() missing 1 required positional argument: 'tracing_context'

```